x-pack/plugins/security_solution/public/detections/components/rules/rule_preview/preview_logs.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Fragment, useMemo } from 'react';
+import { EuiCallOut, EuiText, EuiSpacer, EuiAccordion } from '@elastic/eui';
+import { RulePreviewLogs } from '../../../../../common/detection_engine/schemas/request';
+import * as i18n from './translations';
+
+interface PreviewLogsComponentProps {
+  logs: RulePreviewLogs[];
+  hasNoiseWarning: boolean;
+}
+
+interface SortedLogs {
+  startedAt?: string;
+  logs: string[];
+}
+
+interface LogAccordionProps {
+  logs: SortedLogs[];
+  isError?: boolean;
+}
+
+export const PreviewLogsComponent: React.FC<PreviewLogsComponentProps> = ({
+  logs,
+  hasNoiseWarning,
+}) => {
+  const sortedLogs = useMemo(
+    () =>
+      logs.reduce<{
+        errors: SortedLogs[];
+        warnings: SortedLogs[];
+      }>(
+        (acc, curr) => ({
+          errors:
+            curr.errors.length > 0 // Filter out empty error logs
+              ? [{ startedAt: curr.startedAt, logs: curr.errors }, ...acc.errors]
+              : acc.errors,
+          warnings:
+            curr.warnings.length > 0 // Filter out empty warning logs
+              ? [{ startedAt: curr.startedAt, logs: curr.warnings }, ...acc.warnings]
+              : acc.warnings,
+        }),
+        { errors: [], warnings: [] }
+      ),
+    [logs]
+  );
+  return (
+    <>
+      <EuiSpacer size="s" />
+      {hasNoiseWarning ?? <CalloutGroup logs={[i18n.QUERY_PREVIEW_NOISE_WARNING]} />}
+      <LogAccordion logs={sortedLogs.errors} isError />
+      <LogAccordion logs={sortedLogs.warnings} />
+    </>
+  );
+};
+
+const LogAccordion: React.FC<LogAccordionProps> = ({ logs, isError }) => {
+  const firstLog = logs[0];
+  const restOfLogs = logs.slice(1);
+  return firstLog != null ? (
+    <>
+      <CalloutGroup logs={firstLog.logs} startedAt={firstLog.startedAt} isError={isError} />
+      {restOfLogs.length > 0 ? (
+        <EuiAccordion
+          id={isError ? 'previewErrorAccordion' : 'previewWarningAccordion'}
+          buttonContent={
+            isError ? i18n.QUERY_PREVIEW_SEE_ALL_ERRORS : i18n.QUERY_PREVIEW_SEE_ALL_WARNINGS
+          }
+        >
+          {restOfLogs.map((log, key) => (
```

Comment:
do the logs have a key defined? I thought the reducer above didn't actually add any keys